File: jeongyeon/backjoon/190801/b_2667.cpp
#include <iostream>
#include <cstdio>
#include <algorithm>
#include <vector>
using namespace std;
int input[26][26];
int visit[26][26];
int dx[4]={0, 0, 1, -1};
int dy[4]={1, -1, 0, 0};
vector<int> result;
int N, cnt;

void DFS(int x, int y){
    cnt++;
    visit[x][y] =1;
    for(int i=0; i<4; i++){
        int nx = x +dx[i], ny = y +dy[i];
        if(0 <= nx && nx < N && 0 <= ny && ny < N){
            if (visit[nx][ny] == 0 && input[nx][ny] == 1)
                DFS(nx, ny);
        }
    }
}

int main(){
    cin >> N;
    for(int i=0; i<N; i++){
        for(int j=0; j<N; j++)
            scanf("%1d", &input[i][j]);
    }
    for(int i=0; i<N; i++){
        for(int j=0; j<N; j++){
            if(input[i][j] == 1 && visit[i][j] == 0){
                cnt=0;
                DFS(i, j);
                result.push_back(cnt);
            }
        }   
    }

    sort(result.begin(), result.end());
    cout << result.size() << "\n";
    for(int i=0; i<result.size(); i++)
        cout << result[i] << "\n";

    return 0;
}

File: Other_Lang/b_2667.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;

public class b_2667 {
    public static ArrayList<Integer> result = new ArrayList<Integer>();
    public static int N, cnt;
    static int[][] arr, visited;
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static int[] dx = {0, 1, -1, 0};
    static int[] dy = {1, 0, 0, -1};

    public static void dfs(int x, int y){
        cnt++;
        for(int i=0; i<4; i++){
            int nx = x + dx[i];
            int ny = y + dy[i];
            if(nx < 0 || ny < 0 || nx >= N || ny >= N){ continue; }
            if(arr[nx][ny] == 1 && visited[nx][ny] == 0){
                visited[nx][ny] = 1;
                dfs(nx, ny);
            }
        }
    }

    public static void main(String[] args) throws IOException {
        N = Integer.parseInt(br.readLine());
        arr = new int[N][N];
        visited = new int[N][N];

        for(int i=0; i<N; i++){
            String input = br.readLine();
            for(int j=0; j<N; j++){
                arr[i][j] = Integer.parseInt(input.substring(j, j+1));
            }
        }

        for(int i=0; i<N; i++){
            for(int j=0; j<N; j++){
                if(visited[i][j] == 0 && arr[i][j] == 1){
                    cnt = 0;
                    visited[i][j] = 1;
                    dfs(i, j);
                    result.add(cnt);
                }
            }
        }

        Collections.sort(result);
        System.out.println(result.size());
        for(int i=0; i<result.size(); i++){
            System.out.println(result.get(i));
        }
    }
}